Documentation for model interdependencies (#275)
validmind/documentation: site/guide/model-inventory/configure-model-interdependencies.qmd

---
title: "Configure model interdependencies"
date: last-modified
filters:
  - tachyons
---

Link two or more models in your ValidMind model inventory together. Dependencies are useful to understand a model is
affected by or affects other models in your inventory.

::: {.prereq}

## Prerequisites

- [x] {{< var link.login >}}
- [x] There are two or more models registered in the model inventory.[^1]
- [x] You are the model owner for the model you want to set interdependencies for, a `Developer`, or assigned another role with sufficient permissions to perform the tasks in this guide.[^2]

:::

## View interdependencies

1. In the left sidebar, click **Model Inventory**. 

2. Select a model by clicking on it or find your model by applying a filter or searching for it.[^3]

> 3. Any existing model interdepencies will be shown under the Model Interdepencies section.

Reviewer: Use present tense and spell good: 😛 

```suggestion
3. Any existing model interdependencies are shown under the Model Interdependencies section.
```

## Edit interdependencies

1. In the left sidebar, click **Model Inventory**. 

2. Select a model by clicking on it or find your model by applying a filter or searching for it.[^4]

3. Click **Manage Dependencies** to open up the model interdependencies detail menu.

4. Select either the **{{< fa arrow-up >}} Upstream** or **{{< fa arrow-down >}}Downstream** tab:

    - **Upstream models** — Provides input or intermediate results to another model.
    - **Downstream models** — Receives input and typically performs further processing, predictions, or actions based on that input.

    You can swap between the **{{< fa arrow-up >}} Upstream** or **{{< fa arrow-down >}}Downstream** tabs to set both interdependency types at once.

5. On your list of models populated by the model inventory, add or remove interdependencies as desired: 

    - **Add** — Check off any models that are either upstream or downstream dependent on this current model.
    - **Remove** — Uncheck any models previously linked as upstream or downstream dependent on this current model.

6. Click **Save** to apply your changes.

> Once added, model interdepencies will populate on all models linked together.

Reviewer: Use the same action word for consistency and spell good: 

```suggestion
Once applied, model interdependencies will populate on all models linked together.
```
